build/build.go

```diff
@@ -385,11 +386,13 @@ func toSolveOpt(ctx context.Context, node builder.Node, multiDriver bool, opt Op
 	}
 	defers = append(defers, releaseLoad)
 
-	if sharedKey := so.LocalDirs["context"]; sharedKey != "" {
-		if p, err := filepath.Abs(sharedKey); err == nil {
-			sharedKey = filepath.Base(p)
+	if lm, ok := so.LocalMounts["context"]; ok {
+		if sharedKey, ok := fsutil.GetRoot(lm); ok && sharedKey != "" {
```

Comment:
iiuc this can just be moved inside `LoadInputs` where there is direct access to the value before it is turned into `FS`.

Reply:
There is also `vcs:localdir` to handle. Let me know if this implementation lgty.

Reply:
I think pass the callback that sets the `vcs:localdir` to `toSolveOpt` (and if needed then to `LoadInputs`)